ClassNotFound exception

### Issue submitter TODO list

- [X] I've looked up my issue in [FAQ](https://ui.docs.kafbat.io/faq/common-problems)
- [X] I've searched for an already existing issues [here](https://github.com/kafbat/kafka-ui/issues)
- [X] I've tried running `main`-labeled docker image and the issue still persists there
- [X] I'm running a supported version of the application which is listed [here](https://github.com/kafbat/kafka-ui/blob/main/SECURITY.md)

### Describe the bug (actual behavior)

I ran the kafbat kafka-ui as specified in the following docker-compose [example file](https://github.com/kafbat/ui-serde-glue/blob/main/docker-compose/setup-example.yaml).

The only thing I tweaked is, I downloaded the jar file https://github.com/kafbat/ui-serde-glue/releases/download/1.0.0/kafkaui-glue-serde-1.0.0-jar-with-dependencies.jar and placed it in the  volume mount path`/glue-serde`.

When the application comes up, it fails with the following error

```
kafka-ui      | 	... 42 common frames omitted
kafka-ui      | Caused by: java.lang.ClassNotFoundException: com.provectus.kafka.ui.serde.api.Serde
kafka-ui      | 	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:445)
kafka-ui      | 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:592)
kafka-ui      | 	at io.kafbat.ui.serdes.CustomSerdeLoader$ChildFirstClassloader.loadClass(CustomSerdeLoader.java:122)
kafka-ui      | 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
kafka-ui      | 	... 63 common frames omitted
```
I believe, this is due to referencing `com.provectus.kafka.ui.serde.api.Serde` instead of referencing the kafbat version of this class file at https://github.com/kafbat/kafka-ui/blob/main/serde-api/src/main/java/io/kafbat/ui/serde/api/Serde.java.


### Expected behavior

The application should run succesfully without any errors.

### Your installation details

I used the following docker-compose.yml file to run the application

version: '2'
services:

  kafka-ui:
    container_name: kafka-ui
    image: ghcr.io/kafbat/kafka-ui:latest
    ports:
      - 8080:8080
    depends_on:
      - zookeeper0
      - kafka0
    environment:
      # you can specify aws creds globally. Optional.
      AWS_ACCESS_KEY_ID: '{ AWS_ACCESS_KEY_ID }'
      AWS_SECRET_ACCESS_KEY: '{ AWS_SECRET_ACCESS_KEY }'

      kafka.clusters.0.name: MyFavoriteCluster
      kafka.clusters.0.bootstrapServers: kafka0:29092

      kafka.clusters.0.serde.0.name: GlueSchemaRegistry
      kafka.clusters.0.serde.0.filePath: /glue-serde/kafkaui-glue-serde-1.0.0-jar-with-dependencies.jar
      kafka.clusters.0.serde.0.className: com.provectus.kafka.ui.serdes.glue.GlueSerde
      kafka.clusters.0.serde.0.properties.region: us-east-1  #required
      kafka.clusters.0.serde.0.properties.registry: kui-test #required, name of Glue Schema Registry
      # template that will be used to find schema name for topic key. Optional, default is null (not set).
      kafka.clusters.0.serde.0.properties.keySchemaNameTemplate: "%s-key"
      # template that will be used to find schema name for topic value. Optional, default is  '%s'
      kafka.clusters.0.serde.0.properties.valueSchemaNameTemplate: "%s-value"
      # schema name -> topics pattern where it will be used for keys. Optional.
      kafka.clusters.0.serde.0.properties.topicKeysSchemas.some-topic-key: "some-topic1|some-topic2"
      # schema name -> topics pattern where it will be used for values. Optional.
      kafka.clusters.0.serde.0.properties.topicValuesSchemas.some-topic-value: "some-topic1|some-topic2"
      kafka.clusters.0.serde.0.properties.topicValuesSchemas.another-topic-val: "another-topic-value"

      # you can explicitly specify aws creds on serde level (not globally, Optional.):
      # by providing access keys:
      kafka.clusters.0.serde.0.properties.awsAccessKeyId:  '{ AWS_ACCESS_KEY_ID }'
      kafka.clusters.0.serde.0.properties.awsSecretAccessKey:  '{ AWS_SECRET_ACCESS_KEY }'
      # or by using profile path properties:
      kafka.clusters.0.serde.0.properties.awsProfileName: '{ profile which will be looked in profile file, "default" by default}'
      kafka.clusters.0.serde.0.properties.awsProfileFile: '{ path to profile path, "~/.aws/credentials" by default }'

    volumes:
      - ./:/glue-serde

  zookeeper0:
    image: confluentinc/cp-zookeeper:7.2.1
    environment:
      ZOOKEEPER_CLIENT_PORT: 2181
      ZOOKEEPER_TICK_TIME: 2000
    ports:
      - 2181:2181

  kafka0:
    image: confluentinc/cp-kafka:7.2.1
    depends_on:
      - zookeeper0
    ports:
      - 9092:9092
      - 9997:9997
    environment:
      KAFKA_BROKER_ID: 1
      KAFKA_ZOOKEEPER_CONNECT: zookeeper0:2181
      KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://kafka0:29092,PLAINTEXT_HOST://localhost:9092
      KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: PLAINTEXT:PLAINTEXT,PLAINTEXT_HOST:PLAINTEXT
      KAFKA_INTER_BROKER_LISTENER_NAME: PLAINTEXT
      KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1

### Steps to reproduce

1. Download the https://github.com/kafbat/ui-serde-glue/releases/download/1.0.0/kafkaui-glue-serde-1.0.0-jar-with-dependencies.jar and place it in the same folder as that of the docker-compose file.
2. Ensure the volumes mounted reflect the following path. The jar file is in the same folder as that of docker-compose.yml file and the jar file is thus available to the application container at the `glue-serde` path.
   volumes:
      - ./:/glue-serde
3. docker-compose up

### Screenshots

_No response_

### Logs

redacted

### Additional context

_No response_

--- a/.github/workflows/pr.yaml
+++ b/.github/workflows/pr.yaml
@@ -1,0 +1,28 @@
+name: Pull Request Build
+
+on:
+
+  pull_request:
+    types: [ "opened", "reopened", "synchronize",  ]
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{steps.build.outputs.version}}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'zulu'
+          cache: 'maven'
+
+      - name: Build with Maven
+        id: build
+        run: |
+          mvn -B -V -ntp clean package

--- a/README.md
+++ b/README.md
@@ -1,6 +1,6 @@
 # Glue Schema Registry serde for kafka-ui
 
-This is pluggable serde implementation for [kafka-ui](https://github.com/provectus/kafka-ui/).
+This is pluggable serde implementation for [kafka-ui](https://github.com/kafbat/kafka-ui/).
 
 You can read about Glue Schema [registry](https://docs.aws.amazon.com/glue/latest/dg/schema-registry.html) and how it can be applied for [Kafka usage](https://docs.aws.amazon.com/glue/latest/dg/schema-registry.html).
 

--- a/docker-compose/setup-example.yaml
+++ b/docker-compose/setup-example.yaml
@@ -4,7 +4,7 @@ services:
 
   kafka-ui:
     container_name: kafka-ui
-    image: provectuslabs/kafka-ui:latest
+    image: ghcr.io/kafbat/kafka-ui:latest
     ports:
       - 8080:8080
     depends_on:
@@ -20,7 +20,7 @@ services:
 
       kafka.clusters.0.serde.0.name: GlueSchemaRegistry
       kafka.clusters.0.serde.0.filePath: /glue-serde/kafkaui-glue-serde-1.0-SNAPSHOT-jar-with-dependencies.jar
-      kafka.clusters.0.serde.0.className: com.provectus.kafka.ui.serdes.glue.GlueSerde
+      kafka.clusters.0.serde.0.className: io.kafbat.ui.serde.glue.GlueSerde
       kafka.clusters.0.serde.0.properties.region: us-east-1  #required
       kafka.clusters.0.serde.0.properties.registry: kui-test #required, name of Glue Schema Registry
       # template that will be used to find schema name for topic key. Optional, default is null (not set).

--- a/pom.xml
+++ b/pom.xml
@@ -1,10 +1,13 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
   xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+
   <modelVersion>4.0.0</modelVersion>
-  <groupId>org.example</groupId>
-  <artifactId>kafkaui-glue-serde</artifactId>
+
+  <groupId>io.kafbat.ui.serde</groupId>
+  <artifactId>serde-glue</artifactId>
   <version>1.0-SNAPSHOT</version>
-  <name>Archetype - kafkaui-glue-serde</name>
+
+  <name>kafbat-ui-serde-glue</name>
   <url>http://maven.apache.org</url>
   <packaging>jar</packaging>
 
@@ -56,10 +59,11 @@
       <version>7.0.1</version>
       <scope>provided</scope>
     </dependency>
+
     <dependency>
-      <groupId>com.provectus</groupId>
-      <artifactId>kafka-ui-serde-api</artifactId>
-      <version>1.1.0</version>
+      <groupId>io.kafbat.ui</groupId>
+      <artifactId>serde-api</artifactId>
+      <version>1.0.0</version>
       <scope>provided</scope>
     </dependency>
 

--- a/src/main/java/io/kafbat/ui/serde/glue/GlueSerde.java
+++ b/src/main/java/io/kafbat/ui/serde/glue/GlueSerde.java
@@ -34,7 +34,7 @@ import java.util.function.Function;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 import javax.annotation.Nullable;
-import lombok.NonNull;
+import javax.validation.constraints.NotNull;
 import org.apache.avro.Schema;
 import org.apache.avro.generic.GenericRecord;
 import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
@@ -335,8 +335,8 @@ public class GlueSerde implements Serde {
     private static final FixedJsonDeserializer JSON_DESERIALIZER = new FixedJsonDeserializer();
 
     @Override
-    public GlueSchemaRegistryDataFormatDeserializer getInstance(@NonNull DataFormat dataFormat,
-                                                                @NonNull GlueSchemaRegistryConfiguration configs) {
+    public GlueSchemaRegistryDataFormatDeserializer getInstance(@NotNull DataFormat dataFormat,
+                                                                @NotNull GlueSchemaRegistryConfiguration configs) {
       if (dataFormat == DataFormat.JSON) {
         return JSON_DESERIALIZER;
       } else {
@@ -353,8 +353,8 @@ public class GlueSerde implements Serde {
         GlueSchemaRegistryDeserializerDataParser.getInstance();
 
     @Override
-    public Object deserialize(@NonNull ByteBuffer buffer,
-                              @NonNull com.amazonaws.services.schemaregistry.common.Schema schemaObject) {
+    public Object deserialize(@NotNull ByteBuffer buffer,
+                              @NotNull com.amazonaws.services.schemaregistry.common.Schema schemaObject) {
       String schema = schemaObject.getSchemaDefinition();
       byte[] data = DESERIALIZER_DATA_PARSER.getPlainData(buffer);
       return JsonDataWithSchema.builder(schema, new String(data)).build();

--- a/src/main/java/io/kafbat/ui/serde/glue/GlueSerde.java
+++ b/src/main/java/io/kafbat/ui/serde/glue/GlueSerde.java
@@ -1,4 +1,4 @@
-package com.provectus.kafka.ui.serdes.glue;
+package io.kafbat.ui.serde.glue;
 
 import com.amazonaws.services.schemaregistry.common.AWSDeserializerInput;
 import com.amazonaws.services.schemaregistry.common.GlueSchemaRegistryDataFormatDeserializer;
@@ -15,15 +15,18 @@ import com.google.common.cache.CacheBuilder;
 import com.google.common.cache.CacheLoader;
 import com.google.common.cache.LoadingCache;
 import com.google.protobuf.DynamicMessage;
-import com.provectus.kafka.ui.serde.api.DeserializeResult;
-import com.provectus.kafka.ui.serde.api.RecordHeaders;
 import io.confluent.kafka.schemaregistry.protobuf.ProtobufSchema;
-
+import io.kafbat.ui.serde.api.DeserializeResult;
+import io.kafbat.ui.serde.api.PropertyResolver;
+import io.kafbat.ui.serde.api.RecordHeaders;
+import io.kafbat.ui.serde.api.SchemaDescription;
+import io.kafbat.ui.serde.api.Serde;
 import java.net.URI;
 import java.nio.ByteBuffer;
 import java.nio.file.Path;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 import java.util.UUID;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeUnit;
@@ -31,18 +34,15 @@ import java.util.function.Function;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 import javax.annotation.Nullable;
-
 import lombok.NonNull;
 import org.apache.avro.Schema;
 import org.apache.avro.generic.GenericRecord;
-import software.amazon.awssdk.auth.credentials.*;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.AwsSessionCredentials;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.ProfileCredentialsProvider;
 import software.amazon.awssdk.http.apache.ApacheHttpClient;
-import com.provectus.kafka.ui.serde.api.PropertyResolver;
-import com.provectus.kafka.ui.serde.api.SchemaDescription;
-import com.provectus.kafka.ui.serde.api.Serde;
-
-import java.util.Optional;
-
 import software.amazon.awssdk.profiles.ProfileFile;
 import software.amazon.awssdk.profiles.ProfileFileSystemSetting;
 import software.amazon.awssdk.regions.Region;
@@ -155,10 +155,9 @@ public class GlueSerde implements Serde {
     Optional<String> awsSecretKey = serdeProperties.getProperty("awsSecretAccessKey", String.class);
     Optional<String> awsSessionToken = serdeProperties.getProperty("awsSessionToken", String.class);
     if (awsAccessKey.isPresent() && awsSecretKey.isPresent()) {
-      if (awsSessionToken.isEmpty()) {
-        return () -> AwsBasicCredentials.create(awsAccessKey.get(), awsSecretKey.get());
-      }
-      return () -> AwsSessionCredentials.create(awsAccessKey.get(), awsSecretKey.get(), awsSessionToken.get());
+      return awsSessionToken.<AwsCredentialsProvider>map(
+              s -> () -> AwsSessionCredentials.create(awsAccessKey.get(), awsSecretKey.get(), s))
+          .orElseGet(() -> () -> AwsBasicCredentials.create(awsAccessKey.get(), awsSecretKey.get()));
     }
 
     Optional<String> profileName = serdeProperties.getProperty("awsProfileName", String.class);
@@ -293,7 +292,8 @@ public class GlueSerde implements Serde {
             glueClient.getSchemaVersion(
                 GetSchemaVersionRequest.builder()
                     .schemaId(SchemaId.builder().registryName(registryName).schemaName(schemaName).build())
-                    .schemaVersionNumber(SchemaVersionNumber.builder().versionNumber(schemaResponse.latestSchemaVersion()).build())
+                    .schemaVersionNumber(
+                        SchemaVersionNumber.builder().versionNumber(schemaResponse.latestSchemaVersion()).build())
                     .build())
         );
       } catch (EntityNotFoundException nfe) {
@@ -307,7 +307,8 @@ public class GlueSerde implements Serde {
     return new Deserializer() {
       @Override
       public DeserializeResult deserialize(RecordHeaders recordHeaders, byte[] bytes) {
-        Object obj = deserializationFacade.deserialize(AWSDeserializerInput.builder().buffer(ByteBuffer.wrap(bytes)).build());
+        Object obj =
+            deserializationFacade.deserialize(AWSDeserializerInput.builder().buffer(ByteBuffer.wrap(bytes)).build());
         String val = null;
         if (obj instanceof GenericRecord) {
           val = JsonUtil.avroRecordToJson((GenericRecord) obj);

--- a/src/main/java/io/kafbat/ui/serde/glue/JsonUtil.java
+++ b/src/main/java/io/kafbat/ui/serde/glue/JsonUtil.java
@@ -1,4 +1,4 @@
-package com.provectus.kafka.ui.serdes.glue;
+package io.kafbat.ui.serde.glue;
 
 import com.google.protobuf.DynamicMessage;
 import com.google.protobuf.InvalidProtocolBufferException;

--- a/src/test/java/io/kafbat/ui/serde/glue/GlueSerdeTest.java
+++ b/src/test/java/io/kafbat/ui/serde/glue/GlueSerdeTest.java
@@ -1,4 +1,4 @@
-package com.provectus.kafka.ui.serdes.glue;
+package io.kafbat.ui.serde.glue;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertFalse;
@@ -20,10 +20,10 @@ import com.fasterxml.jackson.databind.json.JsonMapper;
 import com.google.common.base.Preconditions;
 import com.google.protobuf.Descriptors;
 import com.google.protobuf.DynamicMessage;
-import com.provectus.kafka.ui.serde.api.DeserializeResult;
-import com.provectus.kafka.ui.serde.api.PropertyResolver;
-import com.provectus.kafka.ui.serde.api.Serde;
 import io.confluent.kafka.schemaregistry.protobuf.ProtobufSchema;
+import io.kafbat.ui.serde.api.DeserializeResult;
+import io.kafbat.ui.serde.api.PropertyResolver;
+import io.kafbat.ui.serde.api.Serde;
 import java.time.Duration;
 import java.util.ArrayList;
 import java.util.Comparator;
@@ -307,7 +307,7 @@ class GlueSerdeTest {
   @Test
   void testProtoFormatSerdeCompatibility() throws Exception {
     var schema = "syntax = \"proto3\";\n"
-        + "package com.provectus;\n"
+        + "package io.kafbat;\n"
         + "\n"
         + "message TestProtoRecord {\n"
         + "  string field1 = 1;\n"

